### PR TITLE
CYB #79: Stream Client Java IT

### PR DESCRIPTION
**Pull request by ashvayka**

### CODE REVIEW

File: it/pom.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    <profile>
+      <id>it-local</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-failsafe-plugin</artifactId>
+            <configuration>
+              <systemPropertyVariables>
+                <stream_client_it_config>local.conf</stream_client_it_config>
```
Comment (jwang47) [inline it/pom.xml]: I think typically the XML property names we use are `camelCase`, would be better to have that if possible

File: it/stream-client-java-it/src/test/java/co/cask/cdap/client/StreamClientIT.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  public static final String TEST_STREAM = "testStream";
+
+  @Test
+  public void testStreamClient() throws IOException {
+    StreamClient client = getTestClient();
+    //Test that we are able to create a stream
+    client.create(TEST_STREAM);
+
+    //Test that we are able to get/set TTL
+    long ttl = System.currentTimeMillis() + 1000 * 60 * 60 * 24;
```
Comment (jwang47) [inline it/stream-client-java-it/src/test/java/co/cask/cdap/client/StreamClientIT.java]: Can use `long ttl = System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1)` instead

File: it/pom.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    <version>1.0.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>it</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+  <name>CDAP ingest integration tests</name>
+
+  <modules>
+    <module>stream-client-java-it</module>
```
Comment (jwang47) [inline it/pom.xml]: Rename module to `stream-client-java-integration-tests` for clarity

**Comment by jwang47** [comment]
Completed the review, please address the comments.

**Comment by poornachandra** [comment]
@ashvayka This looks good, please run these tests against clusters that @sreevatsanraman provides before merging.

**Comment by ashvayka** [comment]
Fixed review comments

**Comment by ashvayka** [comment]
I have tested on both secure and not secure remote cdap installation

File: it/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+CDAP Ingest Integration Tests
```
Comment (awholegunch) [inline it/README.md]: Remove underline and replace with
#Cask Data Application Platform (CDAP) Ingest Integration Tests

File: it/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+==================
+
+This project contains set of integration tests for various CDAP ingest project components.
+Tests are executed against running CDAP instance. Connection information is configured in properties files.
+This files are stored in resources directory per each sub-module. You can launch same integration tests with different
```
Comment (awholegunch) [inline it/README.md]: These files are stored in the resource directory in each sub-module. You can launch the same integration tests with different configuration file using maven profiles: it-local, it-local-auth, it-remote and it-remote-auth.

File: it/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+==================
+
+This project contains set of integration tests for various CDAP ingest project components.
+Tests are executed against running CDAP instance. Connection information is configured in properties files.
+This files are stored in resources directory per each sub-module. You can launch same integration tests with different 
+configuration file using maven profiles: it-local, it-local-auth, it-remote and it-remote-auth
+
+## Usage
+
+ To launch integration tests against local CDAP instance execute:
```
Comment (awholegunch) [inline it/README.md]: See https://github.com/caskco/cdap-ingest/pull/27/files#diff-1 for the changes I made there, as they are identical here...

File: it/stream-client-java-integration-tests/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+CDAP Stream Client Integration Tests
```
Comment (awholegunch) [inline it/stream-client-java-integration-tests/README.md]: Please see https://github.com/caskco/cdap-ingest/pull/27/files#diff-12 as the changes I made there are identical here.

**Comment by awholegunch** [comment]
Comments added about documentation. These docs look identical to https://github.com/caskco/cdap-ingest/pull/27...

**Comment by ashvayka** [comment]
Fixed comments about documentation

File: it/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+#Cask Data Application Platform (CDAP) Ingest Integration Tests
```
Comment (awholegunch) [inline it/README.md]: # Cask Data Application Platform (CDAP) Ingest Integration Tests

File: it/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+This project contains set of integration tests for various CDAP ingest project components.
+Tests are executed against running CDAP instance. Connection information is configured in properties files.
+These files are stored in the resource directory in each sub-module. You can launch the same integration tests with 
+different configuration file using maven profiles: it-local, it-local-auth, it-remote and it-remote-auth.
```
Comment (awholegunch) [inline it/README.md]: different configuration files

**Comment by awholegunch** [comment]
two minor fixes, then LGTM as far as docs go.

**Comment by ashvayka** [comment]
All doc issues fixed

**Comment by jwang47** [comment]
Please rename the `it` directory to `integration-tests` and then merge.

**Comment by ashvayka** [comment]
Several comments/suggestions from my side:
1. This PR is now dependent on https://github.com/caskco/cdap-clients/pull/10. We need to merge https://github.com/caskco/cdap-clients/pull/10 review and merge it first.
2. I propose to do "it -> integration-tests" renaming in a separate PR, because https://github.com/caskco/cdap-ingest/pull/27 and other items will require refactoring too.
3. Only members with write access to cdap repo can accept pull requests, so i can't do merge.

**Comment by jwang47** [comment]
Sounds good to me, let's merge after https://github.com/caskco/cdap-clients/pull/10 is merged

**Comment by sreevatsanraman** [comment]
Agree - let's rename in a new PR. @ashvayka @jwang47 commented on the other PR - caskco/cdap-clients#10 . Let me know once you have made the change. I will merge both.

**Comment by sreevatsanraman** [comment]
lgtm :+1:

**Pull request merged**